plugins/rbac-backend/src/service/policies-rest-api.ts

```diff
@@ -47,6 +55,9 @@ export class PolicesServer {
     private readonly permissions: PermissionEvaluator,
     private readonly options: RouterOptions,
     private readonly enforcer: Enforcer,
+    private readonly config: Config,
+    private readonly logger: Logger,
+    private readonly discovery: PluginEndpointDiscovery,
```

Comment:
Why don't you add a `pluginIDRetriever` parameter in this constructor, which woukd be a function type returning a list of strings (or an interface if you prefer).
This argument would be passed to the PolicyBuilder by the backend app, and from the policy builder down to here.

That will allow you decoupling the retrieval of IDs from the rest of the logic.
You could even provide a default implementation of this argument (with a  default value).
And for the showcase, we would build this list by using the fixed list of static plugins + the BackendPluginsManager.

Reply:
@davidfestal Thanks, done.